### PR TITLE
[Feature] subscriber module

### PR DESCRIPTION
**Pull request by Megver83**
Hi, some time ago I watched this project and found it pretty useful, idk if you are actively developing it, I'd like to do some contributions. Before getting straight to the point, I'd like to recommend adding a contributors guide, I added a new module and put the copyright under your name. I really don't mind copyrights and credits, but a guide for contributing to this project may be better than not having it.

This PR adds subscription features. The main changes are:

- New class: `NtfySubscriber` which represents the subscriber
- New interfaces (intended for typing): `MessageData` and `Attachment`, which represent the received message and its attachment, if any
- Turn the `message` and `publisher` modules into submodules of the new `publisher` module
- Move `_NtfyURL` to `_urls` module, rename it to `NtfyURL` because it's used by public modules `publisher` and `subscriber`

Minor changes:
- More complete gitignore
- Move `py.typed` under `src/ntfy_api`, otherwise the new interfaces did not work (at least with mypy)
- Add subscription example to the README
- Update dependencies

How subscription works: subscribes to a topic through websockets for better performance, efficiency and lower latency, common httpx client request is used for polling messages.

### CODE REVIEW

**Comment by tanrbobanr** [comment]
Hello Megver83, thanks for contributing! I have in fact been looking to add more features to this, as it was originally designed to suit some immediate needs of mine for a personal project. I will work on getting the repository set up, as well as look over your changes most likely after the holidays, as I am incredibly busy. Although if I find the time, I will try to work on this earlier.

**Comment by Megver83** [comment]
Same here, I also looked at this for a project I plan to develop soon (which requires async programming so my future PRs might be related to that). Other libs where out there but I think this one is the best.

File: src/ntfy_api/publisher/ntfy_publisher.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
     Union,
 )
 from types import MappingProxyType
+from typing import Self
```
Comment (tanrbobanr) [inline src/ntfy_api/publisher/ntfy_publisher.py]: Make sure you do the following for compatibility with versions < 3.11:
```py
if sys.version_info >= (3, 11):
    from typing import Self
else:
    from typing_extensions import Self
```

File: src/ntfy_api/subscriber/ntfy_subscriber.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
     Union,
 )
 from types import MappingProxyType
+from typing import Self
```
Comment (tanrbobanr) [inline src/ntfy_api/subscriber/ntfy_subscriber.py]: Make sure you do the following for compatibility with versions < 3.11:
```py
if sys.version_info >= (3, 11):
    from typing import Self
else:
    from typing_extensions import Self
```

**Comment by tanrbobanr** [comment]
I am going to work on a few changes as well before I release the new version. Thanks again for your contribution!

**Comment by tanrbobanr** [comment]
I have a quick question. I am currently working on getting things integrated since it is now going to be fully-featured. I noticed your `MessageData` class has a `content_type` attribute, which does not appear in [the spec](https://docs.ntfy.sh/subscribe/api/#json-message-format). Is this something you encountered in testing?

**Comment by Megver83** [comment]
I discovered it in testing. When receiving a markdown message, you get a json like this:

```json
{
  "id": "Qvf9bbT4pXRP",
  "time": 1735951725,
  "expires": 1735994925,
  "event": "message",
  "topic": "your-topic",
  "message": "Your message",
  "content_type": "text/markdown"
}
```

**Pull request merged**